code/modules/paperwork/papershredder.dm

```diff
@@ -1,148 +1,252 @@
-		/obj/item/newspaper = 3,
-		/obj/item/card/id = 3,
-		/obj/item/paper_bundle = 3,
-		/obj/item/forensics/sample/print = 1
-		)
+	name               = "paper shredder"
+	desc               = "For those documents you don't want seen."
+	icon               = 'icons/obj/bureaucracy.dmi'
+	icon_state         = "papershredder0"
+	density            = TRUE
+	anchored           = TRUE
+	atom_flags         = ATOM_FLAG_NO_TEMP_CHANGE | ATOM_FLAG_CLIMBABLE
+	obj_flags          = OBJ_FLAG_ANCHORABLE
+	idle_power_usage   = 0
+	stat_immune        = NOSCREEN | NOINPUT
+	waterproof         = FALSE
+	construct_state    = /decl/machine_construction/default/panel_closed
+	required_interaction_dexterity = DEXTERITY_SIMPLE_MACHINES
+	uncreated_component_parts = list(
+		/obj/item/stock_parts/power/apc = 1,
+	)
+	var/list/shredder_bin                                 //List of shreded material type to matter amount
+	var/cached_total_matter  = 0                          //Total of all the matter units we put in the shredder so far
+	var/tmp/max_total_matter = SHEET_MATERIAL_AMOUNT * 10 //Maximum amount of matter that can be stored inside the bin
+
+/obj/machinery/papershredder/Initialize()
+	. = ..()
+	update_icon()
 
-/obj/machinery/papershredder/attackby(var/obj/item/W, var/mob/user)
+/obj/machinery/papershredder/on_component_failure(obj/item/stock_parts/component)
+	. = ..()
 
-	if(istype(W, /obj/item/storage))
-		empty_bin(user, W)
+	if(istype(component, /obj/item/stock_parts/circuitboard))
+		spark_at(get_turf(src), 30, FALSE, src)
+		empty_bin(violent = TRUE)
+
+/**Shreds the given object. */
+/obj/machinery/papershredder/proc/shred(var/obj/item/I, var/mob/user)
+	if(inoperable())
+		to_chat(user, SPAN_WARNING("\The [src] doesn't seem to work."))
 		return
-	else
-		var/paper_result
-		for(var/shred_type in shred_amounts)
-			if(istype(W, shred_type))
-				paper_result = shred_amounts[shred_type]
-		if(paper_result)
-			if(paperamount == max_paper)
-				to_chat(user, "<span class='warning'>\The [src] is full; please empty it before you continue.</span>")
-				return
-			paperamount += paper_result
-			qdel(W)
-			playsound(src.loc, 'sound/items/pshred.ogg', 75, 1)
-			if(paperamount > max_paper)
-				to_chat(user, "<span class='danger'>\The [src] was too full, and shredded paper goes everywhere!</span>")
-				for(var/i=(paperamount-max_paper);i>0;i--)
-					var/obj/item/shreddedp/SP = get_shredded_paper()
-					SP.dropInto(loc)
-					SP.throw_at(get_edge_target_turf(src,pick(global.alldirs)),1,5)
-				paperamount = max_paper
-			update_icon()
-			return
-	..()
-	return
-
-/obj/machinery/papershredder/verb/empty_contents()
-	set name = "Empty bin"
-	set category = "Object"
-	set src in range(1)
-
-	if(usr.incapacitated())
+	if(is_bin_full())
+		visible_message(SPAN_WARNING("The \"bin full\" warning light is flashing on \the [src]!"))
 		return
+	//#TODO: Uncomment this once the bug is fixed, so we check for available power before actually working
```

Comment:
If I recall correctly it was something with can_use_power_oneoff not returning true when there was actually sufficient power on the power net. But its been a while since I wrote that so I might be wrong here.

But from what I remember there was something funky with how power usage is checked for machines that don't have an idle power use, and just pull power one off?